### PR TITLE
[CP-1967] [Files manager](Mac) file extension change in file upload dialog window allows upload of any other file extension

### PR DESCRIPTION
**Pull request by lkowalczyk87**
Jira: [CP-1967]

**Description**

<details>
<summary><b>Screenshots</b></summary>

image

</details>

[CP-1967]: https://appnroll.atlassian.net/browse/CP-1967?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/app/src/files-manager/components/unsupported-file-format/unsupported-file-format.component.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  unsupportedFileFormatUploadModalActionButton: {
+    id: "module.filesManager.unsupportedFileFormatUploadModalActionButton",
+  },
+})
+
+const DuplicatedFilesDetailText = styled(Text)`
+  margin-top: 0.8rem;
+`
+
+const UnsupportedFileFormat: FunctionComponent = ({ ...props }) => {
```
Comment (OskarMichalkiewicz) [inline packages/app/src/files-manager/components/unsupported-file-format/unsupported-file-format.component.tsx:47]: Let's maybe use word "modal" in this component's name as in every other component using ModalDialog.

File: packages/app/src/files-manager/actions/upload-file.action.ts [hunk 2]
```diff
@@ -59,6 +60,18 @@ export const uploadFile = createAsyncThunk(
       return
     }
 
+    const supportedFiles = checkFilesExtensions(filePaths)
```
[the remark this answers is not in the document]
Comment (lkowalczyk87) [reply packages/app/src/files-manager/actions/upload-file.action.ts:63]: any suggestions regarding a new name?

File: packages/app/src/files-manager/actions/upload-file.action.ts [hunk 3]
```diff
@@ -59,6 +60,18 @@ export const uploadFile = createAsyncThunk(
...
+
+    if (!supportedFiles) {
+      dispatch(setUploadBlocked(false))
```
[the remark this answers is not in the document]
Comment (lkowalczyk87) [reply packages/app/src/files-manager/actions/upload-file.action.ts:66]: If the block stays active user will need to restart app to upload another file. uploadBlocked does not represent the case in which upload action has been interrupted

[the remark this answers is not in the document]
Comment (patryk-sierzega) [reply packages/app/src/files-manager/actions/upload-file.action.ts:66]: Understood! :)

File: packages/app/src/files-manager/actions/upload-file.action.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (lkowalczyk87) [reply packages/app/src/files-manager/actions/upload-file.action.ts]: Resolved in cd98e9e979cf9047e310234329283e5050136c46

File: packages/app/src/files-manager/helpers/check-files-extensions.helper.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+ */
+
+import { EligibleFormat } from "App/files-manager/constants/eligible-format.constant"
+
+export const checkFilesExtensions = (filesPaths: string[]): boolean => {
+  return filesPaths.every((filePath) => {
+    return (Object.values(EligibleFormat) as string[]).includes(
+      (filePath.split(".").pop() ?? "").toLocaleLowerCase()
+    )
+  })
```
[the remark this answers is not in the document]
Comment (lkowalczyk87) [reply packages/app/src/files-manager/helpers/check-files-extensions.helper.ts:13]: Resolved in cd98e9e979cf9047e310234329283e5050136c46

**Review by patryk-sierzega — APPROVED** [review]

**Pull request merged**